landoapi/transplants.py

```diff
@@ -301,6 +303,47 @@ def warning_wip_commit_message(*, revision, **kwargs):
         return "This revision is marked as a WIP. Please remove `WIP:` before landing."
 
 
+@RevisionWarningCheck(8, "Repository is under a soft code freeze.", True)
+def warning_code_freeze(*, repo, **kwargs):
+    repo_fields = repo.get("fields", {})
+    short_name = repo_fields.get("shortName")
+    if short_name == "mozilla-central":
+        message = ""
+        try:
+            product_details = requests.get(
+                "https://product-details.mozilla.org/1.0/firefox_versions.json"
+            ).json()
+
+            if "NEXT_SOFTFREEZE_DATE" not in product_details:
+                raise KeyError
+
+            # The code freeze dates generally correspond to PST work days.
+            utc_offset = "-0800"
+            today = datetime.now(tz=timezone.utc)
+            freeze_date = datetime.strptime(
+                f"{product_details.get('NEXT_SOFTFREEZE_DATE')} {utc_offset}",
```

Comment:
No need for `.get` here and below, we should be checking the proper format of these keys before generating the message.